en/03_Drawing_a_triangle/03_Drawing/02_Rendering_and_presentation.md

```diff
@@ -19,46 +19,162 @@ void drawFrame() {
 }
 ```
 
+## Outline of a frame
+
+At a high level, rendering a frame in Vulkan consists of a common set of steps:
+
+* Wait for the previous frame to finish
+* Acquire an image from the swap chain
+* Record a command buffer which draws the scene onto that image
+* Submit the recorded command buffer
+* Present the swap chain image
+
+While we will expand the drawing function in later chapters, for now this is the
+core of our render loop.
+
+<!-- Add an image that shows an outline of the frame -->
+
 ## Synchronization
 
-The `drawFrame` function will perform the following operations:
+<!-- Maybe add images for showing synchronization -->
+
+A core design philosophy in Vulkan is that synchronization of execution on
+the GPU is explicit. The order of operations is up to us to define using various
+synchronization primitives which tell the driver the order we want things to run
+in. This means that many Vulkan API calls which start executing work on the GPU
+are asynchronous, the functions will return before the operation has finished.
+
+In this chapter there are a number of events that we need to order explicitly
+because they happen on the GPU, such as:
 
 * Acquire an image from the swap chain
-* Execute the command buffer with that image as attachment in the framebuffer
-* Return the image to the swap chain for presentation
+* Execute commands that draw onto the acquired image
+* Present that image to the screen for presentation, returning it to the swapchain
 
-Each of these events is set in motion using a single function call, but they are
+Each of these events is set in motion using a single function call, but are all
 executed asynchronously. The function calls will return before the operations
 are actually finished and the order of execution is also undefined. That is
 unfortunate, because each of the operations depends on the previous one
-finishing.
+finishing. Thus we need to explore the which primitives we can use to achieve
+the desired ordering. 
+
+### Semaphores
+
+A semaphore is used to add order between queue operations. Queue operations
+refer to the work we submit to a queue, either in a command buffer or from
+with a function as we will see later. Examples of queues are the graphics queue
+and the presentation queue. Semaphores are used both to order work inside the
+same queue and between different queues. 
+
+There happens to be two kinds of semaphores in Vulkan, binary and timeline.
+Because only binary semaphores will be used in this tutorial, we will not
+discuss timeline semaphores. Further mention of the term semaphore exclusively
+refers to binary semaphores.
+
+A semaphore is either unsignaled or signaled. It begins life as unsignaled. The
```

Comment:
Hmm I see where that's coming from, and is consistent with my understanding of the general concept of semaphores in computer science (like what was taught in my OS class). I'm not sure bringing in a count would help here, since we don't have multiple waiters on a binary semaphore. 

How could this be improved? I am unsure how better to describe the binary semaphore without dropping the CS sync literature book on the reader.

Reply:
I would suggest something along the lines of this:

A binary semaphore can exist in one of two states; signaled and unsignaled.  As the name implies, this binary state is derived in much the same way that a classical semaphore behaves.  The semaphore owns a value known as a count and will signal when the count is non zero.  An example of this behavior is to think of thread memory protection.  Multiple threads can read from memory at the same time, however, each write must be protected from reading at the same time.  This core principal holds true for Vulkan Semaphores, in our example, we choose a very simple use case of a single count which produces a mutual exclusive lock.  In our case, the count is simply 0 or 1, and thus the semaphore is either signaled or not.

Reply:
I still feel like this brings in a lot of unnecessary information. Describing the general CS semaphore brings in a lot of other details, like the count and memory protection from threaded writes.

Reply:
I'm sorry am a little confused, vulkan semaphores are semaphores with a count.  Or am I completely mistaken.  Which has happened before.

Reply:
Vulkan Binary semaphores to me are just a single boolean. Vulkan Timeline semaphores are unlike the classical semaphore description since the uint64_t is always increasing.

So I don't think you are confused, I just see adding descriptions about vulkan binary semaphores being like classical semaphores with a count to only help those who already know about the classical computer science semaphore.

Reply:
That makes sense to me.  Your position is also closer to what the standard advocates.  I think what you have is right as is.  I think technically, it'd be better if we went back in time and advocated that the naming be mutex (mutual exclusion semaphore) as opposed to semaphore and timeline semaphore as semaphore.  Although timeline doesn't allow decrementing the count as a semaphore does those connections should make it easier for reusing the terms without confusion; also timeline allows global access like a semaphore while a mutex is restricted access like the Vulkan semaphore...
I do also wonder if it makes sense to deprecate talking about fences and semaphores and just use timeline semaphores?
I digress, I think it's right as is.  The PR looks good to me.

Reply:
Once timeline semaphores can interact with the swapchain, we could start advocating for that.
I agree that the naming is confusing, hard to not want to go back in time an do a lot of things different in vulkan.

Reply:
:) isn't that the truth!